pkcs11/_pkcs11.pyx

```diff
@@ -927,22 +927,25 @@ class SignMixin(types.SignMixin):
             self.key_type, DEFAULT_SIGN_MECHANISMS,
             mechanism, mechanism_param)
 
+        cdef CK_ULONG session_handle = self.session._handle
+        cdef CK_BYTE *data_buffer = data
+        cdef CK_ULONG data_length = len(data)
         cdef CK_BYTE [:] signature
         cdef CK_ULONG length
 
         with self.session._operation_lock:
-            assertRV(_funclist.C_SignInit(self.session._handle, mech.data, self._handle))
+            assertRV(_funclist.C_SignInit(session_handle, mech.data,
+                                          self._handle))
 
             # Call to find out the buffer length
-            assertRV(_funclist.C_Sign(self.session._handle,
-                            data, <CK_ULONG> len(data),
-                            NULL, &length))
+            assertRV(_funclist.C_Sign(session_handle, &data_buffer[0],
```

Comment:
I wonder if we need this reference-dereference any more. I remember that was mostly to force the typecasting, which should now be sorted on line 931. It's now definitively a `CK_BYTE *`, and so we can probably pass it directly.

Reply:
Although maybe it's better to keep it consistent with `signature`, which needs it still. Hmm.

Reply:
You're right - this isn't needed. I tried it without it and it failed, but that was before I changed the data type to be a pointer rather than using [:]. I'll change it, and the variable name.